discord/utils.py

```diff
@@ -328,12 +333,15 @@ def oauth_url(
         url += f'&permissions={permissions.value}'
     if guild is not MISSING:
         url += f'&guild_id={guild.id}'
-    if redirect_uri is not MISSING:
-        from urllib.parse import urlencode
-
-        url += '&response_type=code&' + urlencode({'redirect_uri': redirect_uri})
     if disable_guild_select:
         url += '&disable_guild_select=true'
+    if redirect_uri is not MISSING or state is not MISSING:
+        from urllib.parse import urlencode
+        
```

Comment:
```suggestion

```

Reply:
Why am I manually looking for whitespace… anyway.

Reply:
lol